cmake/modules/BuildJaeger.cmake

```diff
@@ -22,36 +43,40 @@ function(build_jaeger)
 			-DCMAKE_INSTALL_RPATH_USE_LINK_PATH=TRUE
 			-DOpenTracing_DIR=${CMAKE_SOURCE_DIR}/src/jaegertracing/opentracing-cpp
 			-Dnlohmann_json_DIR=/usr/lib
-			-DCMAKE_INSTALL_PREFIX=${CMAKE_BINARY_DIR}/boost\;${CMAKE_BINARY_DIR}/boost/include\;${CMAKE_BINARY_DIR}/external
-                       -DCMAKE_FIND_ROOT_PATH=${CMAKE_BINARY_DIR}/boost\;${CMAKE_BINARY_DIR}/boost/include\;${CMAKE_BINARY_DIR}/external
+			-DCMAKE_FIND_ROOT_PATH=${CMAKE_BINARY_DIR}/external
+			-DCMAKE_INSTALL_PREFIX=${CMAKE_BINARY_DIR}/external
 			-DCMAKE_INSTALL_LIBDIR=${CMAKE_BINARY_DIR}/external/lib
+			-DBoost_INCLUDE_DIRS=${CMAKE_BINARY_DIR}/boost/include
+                       -DBOOST_ROOT=${CMAKE_BINARY_DIR}/boost
 			-Dthrift_HOME=${CMAKE_BINARY_DIR}/external
 			-DOpenTracing_HOME=${CMAKE_BINARY_DIR}/external)
 
-  set(dependencies opentracing thrift)
+  # build these libraries along with jaeger
+  set(dependencies opentracing)
+  if(NOT WITH_SYSTEM_BOOST)
+    list(APPEND dependencies Boost)
+  endif()
   include(BuildOpenTracing)
   build_opentracing()
-  include(Buildthrift)
-  build_thrift()
-  if(NOT yaml-cpp_FOUND)
-    include(Buildyaml-cpp)
-    build_yamlcpp()
-    add_library(yaml-cpp::yaml-cpp SHARED IMPORTED)
-    add_dependencies(yaml-cpp::yaml-cpp yaml-cpp)
-    set_library_properties_for_external_project(yaml-cpp::yaml-cpp yaml-cpp)
-    list(APPEND dependencies "yaml-cpp")
+  find_package(thrift 0.13.0)
+  if(NOT thrift_FOUND)
+    include(Buildthrift)
+    build_thrift()
+    list(APPEND dependencies thrift)
+    execute_process(COMMAND bash -c "grep -q 'thrift' debian/libjaeger.install || echo 'usr/lib/libthrift.so.*' >> debian/libjaeger.install"
```

Comment:
if we always have `libthrift-0.13.0` installed, why bother adding it to `libjaeger.install`? also, i'd avoid packaging the same file offered by any of the official package in ubuntu focal.

Reply:
it shall only use this if thrift_FOUND is not set, in usual case that will never be the case, but if thrift is not available we need it as a jaeger part of shared libraries needed, again I think while building packages from ceph org this shall never be exercised
https://github.com/ceph/ceph/pull/38783/files#diff-c9968a694423886ad4fa52ca9edfd0c43ca751ea08733b3383d32efc0a14159eR62

Reply:
could you help me understand in what case `thrift_FOUND` is not set and we need to modify `debian/libjaeger.install` ?

Reply:
sure, in local build environment, if someone is not using focal/centos8+ eg. I see a lot of developers still using bionic as dev environment, for them the already worked buildthrift method could be useful. The rest removed dependencies are not as sensitive to version as thrift, I would not want them to struggle with building jaeger

Reply:
why do we need to support bionic? if that's a must we would have to revert all the changes which depend on focal.

Reply:
it's not a must, just a good to have. I wanted to keep this one for just developers who uses vstart for bionic, but we can drop `debian/libjaeger.install` mangling since for shipping package, we would want to support focal

Reply:
i insist. as i don't want to have a "partially supported" cmake script. it would be a recipe of pain in the long run. the first issue for those who build ceph on bionic would be the unsupported cmake version. not to mention other issues, like ancient GCC. if there are a lot of developers using bionic, i'd suggest them either build ceph in a container or upgrade their distro.

Reply:
thanks for your perspective @tchaikov! I would love to move to focal altogether as well! :) 
Completely understand the pains as well dropped all bionic related build needed: